sklearn/metrics/tests/test_classification.py

```diff
@@ -879,12 +879,6 @@ def test_confusion_matrix_dtype():
     assert cm[0, 0] == 4294967295
     assert cm[1, 1] == 8589934590
 
-    # np.iinfo(np.int64).max should cause an overflow
```

Comment:
There might be a neater solution than removing this, but it turns out that different implementations will give different results in the case of overflow

Reply:
Should it just test for the binary type and exact selection conditions as in the optimized code before making the assertions to exclude them?